### PR TITLE
feat: Lydian airdrop

### PR DESCRIPTION
**Pull request by nieldeckx**

### CODE REVIEW

**Review by philipdesmedt — COMMENTED** [review]
Just did a quick review, looks good

1. Should we just send the LDN into the `lydian-airdrop-v1-1` contract when we receive them from the Lydian team?
2. Have we decided on the weights of stDIKO vs DIKO/USDA vs DIKO/STX already? I think DIKO/USDA should have the highest reward weight followed by DIKO/STX followed by stDIKO (from higher risk & capital needed to provide liquidity to lower)

**Comment by nieldeckx** [comment]
1. Yes
2. Weights not implemented yet. Agree with your thoughts.

Have to refactor first as the read calls are too heavy..

**Comment by nieldeckx** [comment]
Added contract `lydian-airdrop-snapshot-v1-1` to get DIKO amounts at given block:
- DIKO/USDA pool
- wSTX/DIKO pool
- stDIKO pool
- Total DIKO supply
We can decide how to split LDN tokens across the pools, and update the constants in `lydian-airdrop-v1-1`.

Had to split up claiming across the pools to reduce transaction costs. 

@philipdesmedt can you review again please?

**Review by philipdesmedt — APPROVED** [review]
Fantastic!

Haven't tried the front-end FYI

File: clarity/contracts/lydian-airdrop-v1-1.clar [hunk 1]
```diff
@@ -0,0 +1,204 @@
+(define-constant ERR-NOT-AUTHORIZED u444001)
+(define-constant ERR-EMERGENCY-SHUTDOWN-ACTIVATED u444002)
+
+;; TODO - SET FOR MAINNET
+(define-constant ldn-for-stdiko u600000000)
```
[the remark this answers is not in the document]
Comment (nieldeckx) [reply clarity/contracts/lydian-airdrop-v1-1.clar:5]: Numbers are incorrect. We still need to define the amount of LDN we will drop per pool.

File: clarity/contracts/lydian-airdrop-v1-1.clar [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+(define-public (emergency-withdraw-tokens)
+  (let (
+    (guardian tx-sender)
+    (contract-balance (unwrap-panic (contract-call? .lydian-token get-balance (as-contract tx-sender))))
+  )
+    (asserts! (is-eq tx-sender (contract-call? .arkadiko-dao get-guardian-address)) (err ERR-NOT-AUTHORIZED))
+
+    ;; TODO - SET MAINNET ADDRESS FOR LYDIAN TOKEN
+    (as-contract (contract-call? .lydian-token transfer contract-balance (as-contract tx-sender) guardian none))
```
[the remark this answers is not in the document]
Comment (nieldeckx) [reply clarity/contracts/lydian-airdrop-v1-1.clar:38]: True :)

**Comment by nieldeckx** [comment]
Webpage:
Screenshot 2022-03-11 at 13 21 01

@Macxim can you have a look to make it a bit nicer?

Clarinet integrate is not working here, Hiro is fixing the bug. You can use mocknet using the `ts-node deploy-contracts.ts` in `web/scripts` folder. Page available at: http://localhost:9000/lydian-airdrop

**Pull request merged**